src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,127 @@
 import fr.esgi.al5_2.Tayarim.dto.auth.AuthLoginResponseDTO;
+import fr.esgi.al5_2.Tayarim.dto.proprietaire.AdministrateurDTO;
 import fr.esgi.al5_2.Tayarim.dto.proprietaire.ProprietaireDTO;
-import fr.esgi.al5_2.Tayarim.entities.Proprietaire;
-import fr.esgi.al5_2.Tayarim.exceptions.*;
+import fr.esgi.al5_2.Tayarim.exceptions.AdministrateurNotFoundException;
+import fr.esgi.al5_2.Tayarim.exceptions.ProprietaireNotFoundException;
+import fr.esgi.al5_2.Tayarim.exceptions.TokenExpireOrInvalidException;
+import fr.esgi.al5_2.Tayarim.exceptions.UtilisateurNotFoundException;
 import lombok.NonNull;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.AbstractMap;
+import java.util.Map.Entry;
 import java.util.UUID;
 
 @Service
 @Transactional(readOnly = true)
 public class AuthService {
 
-    private final ProprietaireService proprietaireService;
-    private final JwtHelper jwtHelper;
+  private final ProprietaireService proprietaireService;
+  private final AdministrateurService administrateurService;
+  private final JwtHelper jwtHelper;
+
+  private final TokenCacheService tokenCacheService;
+
+  public AuthService(ProprietaireService proprietaireService, AdministrateurService administrateurService, JwtHelper jwtHelper, TokenCacheService tokenCacheService) {
+    this.proprietaireService = proprietaireService;
+    this.administrateurService = administrateurService;
+    this.jwtHelper = jwtHelper;
+    this.tokenCacheService = tokenCacheService;
+  }
+
+  @Transactional(readOnly = true, propagation = Propagation.NOT_SUPPORTED)
+  public AuthLoginResponseDTO login(@NonNull String email, @NonNull String password) {
+    ProprietaireDTO proprietaireDTO = null;
+    AdministrateurDTO administrateurDTO = null;
+    Long id;
+    boolean isAdmin = false;
+    try {
+      proprietaireDTO = proprietaireService.getProprietaireByEmail(email);
+      id = proprietaireDTO.getId();
+    } catch (Exception e) {
+      try {
+        administrateurDTO = administrateurService.getAdministrateurByEmail(email);
+        id = administrateurDTO.getId();
+        isAdmin = true;
+      } catch (Exception exception) {
+        throw new UtilisateurNotFoundException();
+      }
 
-    private final TokenCacheService tokenCacheService;
+    }
 
-    public AuthService(ProprietaireService proprietaireService, JwtHelper jwtHelper, TokenCacheService tokenCacheService) {
-        this.proprietaireService = proprietaireService;
-        this.jwtHelper = jwtHelper;
-        this.tokenCacheService = tokenCacheService;
+    if (proprietaireDTO != null && !proprietaireService.verifyPassword(password, proprietaireDTO.getId())) {
+      throw new ProprietaireNotFoundException();
+    } else if (administrateurDTO != null && !administrateurService.verifyPassword(password, administrateurDTO.getId())) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 121).